edge-util/src/Microsoft.Azure.Devices.Edge.Util/edged/version_2018_06_28/generatedCode/HttpWorkloadClient.cs

```diff
@@ -51,32 +130,32 @@ public System.Threading.Tasks.Task<SignResponse> SignAsync(string api_version, s
         {
             return SignAsync(api_version, name, genid, payload, System.Threading.CancellationToken.None);
         }
-
+    
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
         /// <param name="api_version">The version of the API.</param>
         /// <param name="name">The name of the module on whose behalf the payload will be signed. (urlencoded)</param>
         /// <param name="genid">The generation identifier for the module as generated by IoT Hub.</param>
         /// <param name="payload">The data to be signed.</param>
         /// <returns>Ok</returns>
         /// <exception cref="IoTEdgedException">A server side error occurred.</exception>
-        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
         public async System.Threading.Tasks.Task<SignResponse> SignAsync(string api_version, string name, string genid, SignRequest payload, System.Threading.CancellationToken cancellationToken)
         {
             if (name == null)
                 throw new System.ArgumentNullException("name");
-
+    
             if (genid == null)
                 throw new System.ArgumentNullException("genid");
-
+    
             if (api_version == null)
                 throw new System.ArgumentNullException("api_version");
-
+    
             var urlBuilder_ = new System.Text.StringBuilder();
             urlBuilder_.Append(BaseUrl != null ? BaseUrl.TrimEnd('/') : "").Append("/modules/{name}/genid/{genid}/sign?");
-            urlBuilder_.Replace("{name}", System.Net.WebUtility.UrlEncode(ConvertToString(name, System.Globalization.CultureInfo.InvariantCulture)));
-            urlBuilder_.Replace("{genid}", System.Net.WebUtility.UrlEncode(ConvertToString(genid, System.Globalization.CultureInfo.InvariantCulture)));
-            urlBuilder_.Append("api-version=").Append(System.Net.WebUtility.UrlEncode(ConvertToString(api_version, System.Globalization.CultureInfo.InvariantCulture))).Append("&");
+            urlBuilder_.Replace("{name}", System.Uri.EscapeDataString(ConvertToString(name, System.Globalization.CultureInfo.InvariantCulture)));
```

Comment:
it needs to use System.Net.WebUtility.UrlEncode to match with the sdk from what I remember. The encoding has some small differences. There is a note at the beginning of the file, but hard to see it:
Note: Code manually changed to replace System.Uri.EscapeDataString with System.Net.WebUtility.UrlEncode

Reply:
I'll try reverting

Reply:
Let's make the note more obvious. Add a test to catch System.Net.WebUtility.UrlEncode